pkg/util/shell/shell_helper.go

```diff
@@ -34,7 +34,7 @@ func ExecuteWithEnvAndInput(t test.TestHelper, env []string, cmd string, input s
 	t.T().Helper()
 	output, err := execShellCommand(cmd, env, input)
 	if err != nil {
-		t.Fatalf("Command failed: %s\n%serror: %s", cmd, appendNewLine(output), err)
+		t.Logf("Command failed: %s\n%serror: %s", cmd, appendNewLine(output), err)
```

Comment:
This now prevents tests from failing when the command fails. If you need to check the command's output even if the command fails, we need to find a better way to achieve this. We do have some other places where we expect the command to fail, so this is clearly a valid use-case. We just need to find a good way to tell this function whether to fail on command failures or not.

Reply:
Why would it be useless? It's being used in basically all tests. In 99% of cases, if the command fails, the test should fail immediately and not on the next assertion. Not failing fast is one of the worst things in software, as it almost always leads to a lot of wasted time.

Reply:
Because for some errors kubectl returns error and for others it does not. For example, when I execute `kubectl -n metallb-system...` and the namespace does not exist, I get fatal error. If the namespace exist, but a deployment does not exist, I don't get error. :exploding_head:

Reply:
I am not able to verify all the tests for now, so I will revert this change and will use `exec.Command`.

Reply:
```
=== Failed
=== FAIL: pkg/tests/ossm-federation  (0.00s)
    test_helper_setup.go:60: Check if MetalLB operator already exists
    test_helper_setup.go:64: FATAL: Command failed: oc -n metallb-system get deployments/metallb-operator-controller-manager
        Error from server (NotFound): namespaces "metallb-system" not found
        error: exit status 1
panic: FailNow

```

Reply:
Done